lib/modules/manager/bazel-module/stack.ts

```diff
@@ -0,0 +1,22 @@
+export class Stack<T> extends Array<T> {
+  static create<T>(...items: Array<T>): Stack<T> {
+    const stack = new Stack<T>();
+    stack.push(...items);
+    return stack;
+  }
+
+  get safeCurrent(): T | undefined {
+    if (!this.length) {
+      return undefined;
+    }
+    return this[this.length - 1];
+  }
+
+  get current(): T {
+    const c = this.safeCurrent;
+    if (c === undefined) {
+      throw new Error('Requested current, but no value.');
+    }
+    return c;
+  }
```

Comment:
Can't we move those two properties to private props inside `Ctx` class, so we don't need to create an extra object? Would safe use one more object creation inside the parser callbacks.